src/collections/components/AddLabelFormConnector/AddLabelFormConnector.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useState } from 'react';
+import { useRunMutation } from '../../../_shared/hooks';
+import { AddLabelForm } from '../';
+import { FormikHelpers, FormikValues } from 'formik';
+import { useCreateLabelMutation } from '../../../api/generatedTypes';
+
+interface AddLabelFormConnectorProps {
+  /**
+   * Toggle the AddLabelFormConnectorProps to show/hide as necessary.
+   */
+  toggleModal: VoidFunction;
+
+  /**
+   * A helper function from Apollo Client that triggers a new API call to refetch
+   * the data for a given query.
+   */
+  refetch: VoidFunction;
+}
+
+/**
+ * Parent component for the AddLabelForm component
+ */
+export const AddLabelFormConnector: React.FC<AddLabelFormConnectorProps> = (
+  props
+): JSX.Element => {
+  const { toggleModal, refetch } = props;
+  const [isLoaderShowing, setIsLoaderShowing] = useState<boolean>(false);
+  const [createLabelMutation] = useCreateLabelMutation();
+
+  // Get a helper function that will execute each mutation, show standard notifications
+  // and execute any additional actions in a callback
+  const { runMutation } = useRunMutation();
+
+  const onSubmit = (
+    values: FormikValues,
+    formikHelpers: FormikHelpers<any>
+  ): void => {
+    // Run the mutation
+    runMutation(
+      createLabelMutation,
+      { variables: { name: values.labelName } },
+      `"${values.labelName} has been created"`,
+      () => {
+        setIsLoaderShowing(false);
+        toggleModal();
+        formikHelpers.setSubmitting(false);
+
+        if (refetch) {
+          refetch();
+        }
```

Comment:
I think we can remove this and just pass the `refetch` callback as the last parameter for `runMutation()` call. That exact same logic exists in the `runMutation()`

Reply:
ah good catch, missed that